packages/block-library/src/query/edit/inspector-controls/per-page-control.js

```diff
@@ -0,0 +1,24 @@
+/**
+ * WordPress dependencies
+ */
+import { RangeControl } from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
+
+const MIN_POSTS_PER_PAGE = 1;
+const MAX_POSTS_PER_PAGE = 100;
+
+const PerPageControl = ( { perPage, offset = 0, onChange } ) => {
+	return (
+		<RangeControl
+			label={ __( 'Posts Per Page' ) }
+			min={ MIN_POSTS_PER_PAGE }
+			max={ MAX_POSTS_PER_PAGE }
+			onChange={ ( newPerPage ) => {
+				onChange( { perPage: newPerPage, offset } );
+			} }
+			value={ parseInt( perPage, 10 ) }
```

Comment:
I think the `perPage` value can be `null`, resulting in `NaN`.

Reply:
> I think the `perPage` value can be `null`, resulting in `NaN`.

Yes, I noticed in the current implementation on trunk [these values do have checks](https://github.com/WordPress/gutenberg/blob/31a3c5d35f49b1051ca240d8106b305aa6408ccf/packages/block-library/src/query/edit/query-toolbar.js#L50-L56) for `null` and 'out of range' values. But this applies to all the three settings. Not sure why those checks have been removed in this PR but yes they should likely be added back, maybe by using an helper function. 

More work for next week 🍹

Reply:
I've added these back now.